src/main/kotlin/no/nav/familie/ks/sak/kjerne/beregning/BeregningService.kt

```diff
@@ -0,0 +1,23 @@
+package no.nav.familie.ks.sak.kjerne.beregning
+
+import no.nav.familie.ks.sak.kjerne.beregning.domene.AndelTilkjentYtelseRepository
+import no.nav.familie.ks.sak.kjerne.personident.Aktør
+import no.nav.familie.ks.sak.kjerne.personopplysninggrunnlag.domene.PersonopplysningGrunnlagRepository
+import org.springframework.stereotype.Service
+
+@Service
+class BeregningService(
+    private val andelTilkjentYtelseRepository: AndelTilkjentYtelseRepository,
+    private val personopplysningGrunnlagRepository: PersonopplysningGrunnlagRepository
+) {
+
+    /**
+     * Henter alle barn på behandlingen som har minst en periode med tilkjentytelse.
+     */
+    fun finnBarnFraBehandlingMedTilkjentYtelse(behandlingId: Long): List<Aktør> {
+        val andelerTilkjentYtelse = andelTilkjentYtelseRepository.finnAndelerTilkjentYtelseForBehandling(behandlingId)
+
+        return personopplysningGrunnlagRepository.findByBehandlingAndAktiv(behandlingId)?.barna?.map { it.aktør }
```

Comment:
Ser at `finnBarnFraBehandlingMedTilkjentYtelse` kalles ved en revurdering eller teknisk endring.
Da sendes den tidligere behandling.id ned. Hva er da sjansen for at `findByBehandlingAndAktiv(behandlingId)` returnerer null?

Reply:
Støttes 👍

Reply:
Det kan være flere personopplysningsgrunnlag for en behandling. Derfor henter jeg aktiv personopplysninggrunnlag

Reply:
fikset det